### ISSUE TITLE
Enforcing API contract between requester and responder

### ISSUE THREAD

**Issue by nick13jaremek** [dashersw/cote#151]
Hello!

Currently I am implementing a CoteJS requester-responder communication, with two methods exposed on the responder that the requester can call to.

However, I would like to enforce an API contract between the two entities in such a way that:

1. The list of methods a responder exposes (and are callable by a requester) are defined in a configuration-like object.
2. When a requester calls a method on a responder, and such method is not exposed by the responder, an error is raised. Alternatively, the responder could raise an error during its initialization if the methods it has defined do not match the provided list of methods that are exposable.

For this purpose, I saw in the documentation two keys that could suit this use case, namely `respondsTo` on the `Responder` object, and `requestsTo` on the `Requester` object.

However, it seems these two configuration keys are not used at all internally by CoteJS. Instead they seem to exist only for documenting purposes on the code, but have no attached checking logic.

Is there any way to enforce a `Requester/Responder` to implement a list of methods defined in some list, with such list passed to the `Requester/Responder`'s initial configuration? 

Thank you very much in advance!

**Comment by nick13jaremek** [comment]
I just noticed that the `Responder` object contains a key named `listenerTree` whose value is an object of the event names and the functions associated to them. This could be used to check the API contract requirements I outlined in my previous comment.

Also, since the `Responder` and `Requester` objects extend the `EventEmitter2` object, the `EventEmitter2`'s `eventNames` function should return the list of events defined on the `Responder` or `Requester`, but surprisingly it always returns an empty list when called on the `Responder` or the `Requester` instance.

**Comment by dashersw** [comment]
Hi! I don't think the Requester has a `requestsTo` property. `respondsTo` is used only in the Sockend component, for exposing a select group of messages to the frontend.

There are multiple ways to implement this in cote, either through the EventEmitter or overriding certain methods like `on` or `send`. Advertisement object that is passed when initializing a component is actually broadcast to all other services, so every service knows what others could `respondsTo`.

**Comment by nick13jaremek** [comment]
@dashersw Indeed, the `Requester`has no `requestsTo` method nor property, it's something I made up by accident when looking at the `respondsTo` method of the `Responder` component.

Thank you for the descriptions on the potential workarounds to overcome this issue. I ended up attaching a single event handler to the `Responder` instance, and checking in this handler's logic whether the incoming request is valid based on a list of methods provided in a configuration file.

The name of the method to be checked is provided in the Cote request's payload along with its associated data parameters. If the name of the method does not match any of the provided configuration file, then the `Responder` instance returns a Promise rejection which is handled by the `Requester`.

If, on the other hand, the method's name is valid, the responder returns the result of the function that is mapped to the method's name.
